src/platforms/hosted/cmsis_dap.c

```diff
@@ -425,33 +425,34 @@ static void dap_mem_read(adiv5_access_port_s *ap, void *dest, uint32_t src, size
 	if (len == 0)
 		return;
 	align_e align = MIN(ALIGNOF(src), ALIGNOF(len));
-	DEBUG_WIRE("memread @ %" PRIx32 " len %ld, align %d , start: \n", src, len, align);
-	if (((unsigned)(1 << align)) == len)
+	DEBUG_WIRE("dap_mem_read @ %" PRIx32 " len %zu, align %d\n", src, len, align);
+	/* If the read can be done in a single transaction, use the dap_read_single() fast-path */
+	if ((1U << align) == len)
 		return dap_read_single(ap, dest, src, align);
-	/* One word transfer for every byte/halfword/word
-	 * Total number of bytes in transfer*/
-	unsigned int max_size = ((dbg_get_report_size() - 6U) >> (2U - align)) & ~3U;
-	while (len) {
-		dap_ap_mem_access_setup(ap, src, align);
-		/* Calculate length until next access setup is needed */
-		unsigned int blocksize = (src | 0x3ffU) - src + 1U;
-		if (blocksize > len)
-			blocksize = len;
-		while (blocksize) {
-			unsigned int transfersize = blocksize;
-			if (transfersize > max_size)
-				transfersize = max_size;
-			if (!dap_read_block(ap, dest, src, transfersize, align)) {
+	/* Otherwise proceed blockwise */
+	dap_ap_mem_access_setup(ap, src, align);
```

Comment:
This needs to be called inside the loop with `src + offset` to update AP_TAR at 1024B boundaries, otherwise it'll wrap.

Reply:
This should now be fixed - many thanks :)